easylist.go

```diff
@@ -100,25 +107,46 @@ type list struct {
 }
 
 func (l *list) Allow(req *http.Request) bool {
-	dm := l.getMatcher(req.Host)
-	if dm == nil {
+	dms := l.getMatchers(req.Host)
+	if dms == nil {
 		// Until we've been initialized, allow everything
 		return true
 	}
+
+	origin := domainFromURL(req.Header.Get("Origin"))
+	referrer := origin
+	if referrer == "" {
+		referrer = domainFromURL(req.Header.Get("Referer"))
+	}
+
 	ar := &adblock.Request{
-		URL:    req.URL.String(),
-		Domain: req.Host,
+		URL:          req.URL.String(),
+		Domain:       origin,
```

Comment:
Check https://github.com/pmezard/adblock/blob/master/adblock/rules.go#L503 to see how `Domain` field is used, which is different from [the doc](https://github.com/pmezard/adblock/blob/master/adblock/rules.go#L19)